duckduckgo/iOS
# Flatten TabViewController and WebViewController inheritance

Task/Issue URL:
https://app.asana.com/0/276630244458377/542099048519218

**Description**:
This PR merges the TabViewController and WebViewController subclasses to make code easier to understand and improve in the future. This change also
- flattens overridden userAgent method removing unused behavior and making legitimate behavior clear
- combines duplicated url and loadedUrl
- combines confusing UIGestrureRecognizerDelegate methods across classes
- removes repeated code especially around delegate callbacks and navigation
- moves some utilities into extension class
- moves menu builders into extensions to declutter the TabViewController and make it easier to see which option belongs to which menu

**Note:** The TabViewConroller class is quite large now (a bit over 850 lines). While it is larger it should hopefully also be easier to work with. In the future, we will pull out components to further improve it however this goes beyond the scope of this PR. 

**Steps to test this PR**:
Test everything! Make sure all browsing features still work in particular
1. Scrolling
1. Showing and hiding the bars
1. Long press menus, check all the options work too
1. Buttons including forward/back/refresh
1. Desktop mode

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
